assets/scripts/util/fetch_blocking.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Create blocking fetch requests.
+ * Display a blocking shield when a request is in progress.
+ * Only one of these should be active at any time and
+ * no other fetch / xhr / ajax should exist when they do.
+ */
+import $ from 'jquery'
+import { showBlockingShield, hideBlockingShield, darkenBlockingShield } from '../app/blocking_shield'
+
+let blockingAjaxRequest
+let blockingAjaxRequestDoneFunc
+let blockingAjaxRequestCancelFunc
+
+// This needs to be exported to window for streets/xhr (temp?)
+export let blockingAjaxRequestInProgress = false
```

Comment:
Exported variable are statically binded. We need to create and export getter and setter here.

You can see, how it works here: https://gist.github.com/magul/d2ffa5fce8181b97e9279c77c44c31a5